blueman/bluez/Base.py

```diff
@@ -11,30 +11,24 @@
 
 class BaseMeta(GObjectMeta):
     def __call__(cls, *args: str, **kwargs: str) -> "Base":
-        instances: Dict[str, Dict[str, "Base"]] = cls.__dict__.setdefault("__instances__", {})
-
-        path = None
-        interface_name = None
+        if args:
+            raise RuntimeError("Only keyword arguments allowed")
 
-        if kwargs:
-            interface_name = kwargs.get('interface_name')
-            path = kwargs.get('obj_path')
+        if not hasattr(cls, "__instances__"):
+            cls.__instances__: Dict[str, Dict[str, "Base"]] = {}
 
-        if args:
-            for arg in args:
-                if args is None:
-                    continue
-                elif '/' in arg:
-                    path = arg
-                elif '.' in arg:
-                    interface_name = arg
-
-        if not interface_name:
+        # cls._interface_name or cls._obj_path need to exist to act as fallback, they may not
+        interface_name = kwargs.get('interface_name', cls._interface_name)
+        if interface_name is None:
```

Comment:
> Why? This can only be true if `interface_name=None` got provided explicitly which is not expected.

interface_name is pretty much guaranteed not to be in the kwargs passed to ``__call__``.

> Actually no concrete class accepts an `interface_name` keyword at all, so we could just use `cls._interface_name` in any case

Currently it is always provided to ``__init__`` with ``self._interface_name`` so in theory we could just use ``self._interface_name`` and remove it from ``__init__``. I can do that if this is what you  want.

The issues is not so much with interface_name being in kwargs but that cls._interface_name __must__ exist to be used as second argument to ``dict.get()`` otherwise we end up with a ``NameError``. You are correct that this currently is not the case for interface_name, it may never, but it might be true in the future so I made sure to handle this.

Reply:
As `interface_name` is not an argument of any of the implementations we could just use `cls.__instances__` instead of `cls.__instances__[interface_name]` but I'm fine with keeping this unused mechanism if you like.

Reply:
I'll drop interface_name from ``__init__`` in a follow up commit.